cmd/minishift/cmd/openshift/add.go

```diff
@@ -0,0 +1,94 @@
+/*
+Copyright (C) 2018 Red Hat, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package openshift
+
+import (
+	"fmt"
+	"github.com/docker/machine/libmachine"
+	"github.com/docker/machine/libmachine/provision"
+	cmdUtil "github.com/minishift/minishift/cmd/minishift/cmd/util"
+	"github.com/minishift/minishift/cmd/minishift/state"
+	"github.com/minishift/minishift/pkg/minikube/cluster"
+	"github.com/minishift/minishift/pkg/minikube/constants"
+	"github.com/minishift/minishift/pkg/minishift/clusterup"
+	minishiftConstants "github.com/minishift/minishift/pkg/minishift/constants"
+	openshiftVersion "github.com/minishift/minishift/pkg/minishift/openshift/version"
+	"github.com/minishift/minishift/pkg/util/os/atexit"
+	"github.com/spf13/cobra"
+	"os"
+)
+
+const (
+	componentFlag             = "component"
+	nonExistentComponentError = "You need to specify a component name using -c <component_name>"
```

Comment:
not sure why we would need the `-c` option to specify a component

Reply:
The error message looks confusing because instead of saying `you are trying to add a non existent component` we are saying  `You need to specify a component name using -c <component_name>`